EIPS/eip-7336.md

```diff
@@ -0,0 +1,104 @@
+
+### Typical Flow
+
+![Typical Flow](../assets/eip-7336/workflow.png)
+
+### Interface
+
+A new UDoN interface is defined, consisting of the following method:
+
+```solidity
+interface IERC7336 {
+  struct Result {
+      address   resAddr;
+      string    resText;
+      bytes32   resAbi;
+      bytes32   resPublickey;
+  }
+
+  /// @notice Returns the multi type data by retrieving value from different registries
+  /// @param registryAddress An address of target registry
+  /// @param node A key to lookup data for
+  /// @return The value is retrieved from specific registry
+  function getValue(address registryAddress, byte32 node) external view returns (Result res);
+}
+```
+
+Smart contracts implementing the standard MUST implement all of the functions in the interface.
+
+## Rationale
+
+The rationale behind this EIP is to augment the efficiency and interoperability within the Ethereum ecosystem. It provides a seamless way to interact with different registries, ultimately simplifying the process of reading and retrieving data. This proposed interface will encourage the development of more efficient, interoperable and standardized solutions, leading to a more robust and integrated Ethereum ecosystem.
+
+## Backwards Compatibility
+
+As this EIP introduces a new feature and does not modify any existing behaviors, there are no backwards compatibility issues.
+
+## Reference Implementation
+
+```solidity
+pragma solidity ^0.8.12;
+
+import "./IERC7336Interface.sol";
+import "./ERC165/ERC165Query.sol";
+
+contract RegistryInterfaceId {
+  bytes4 internal constant ENS_INTERFACE_ID = 0x75b24222;
+  bytes4 internal constant TXT_INTERFACE_ID = 0x59d1d43c;
+  bytes4 internal constant OTHER_INTERFACE_ID = 0x...; // Replace with the correct value for the other interface ID
+}
+
+contract UDoN is ERC165Query, IERC7336Interface {
+    function getValue(address registryAddress, bytes32 node) external view override returns (Result memory) {
+        UDoN reg = UDoN(registryAddress);
+        Result memory res;
+
+        if (doesContractImplementInterface(registryAddress, RegistryInterfaceId.ENS_INTERFACE_ID)) {
+            var resolver = reg.resolver(node);
+            var address = resolver.addr(node);
+            res.resAddr = address;
+        } else if (doesContractImplementInterface(registryAddress, RegistryInterfaceId.TXT_INTERFACE_ID)) {
+            var text = reg.getText();
+            res.resText = text;
+        }
+        
+        return res;
+    }
+}
+```
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
If you include an HTML-style comment, the linter will make sure you replace it before moving to Review:

```suggestion
Needs discussion. <!-- TODO -->
```